hyperspy/tests/signals/test_ragged_signal.py

```diff
@@ -89,6 +89,19 @@ def test_slicing(self):
         with pytest.raises(RuntimeError):
             s.isig[0]
 
+    def test_ragged_0D_map(self):
+        def return_value(val):
+            return val
+
+        zeroDragged = self.s.inav[0, 0]
+
+        new_signal = zeroDragged.map(return_value, inplace=False)
+        if self.s._lazy:
+            new_signal.compute()
+            assert self.s.data[0, 0].compute().shape == new_signal.data[0].shape
+        else:
+            self.s.data[0, 0].shape == new_signal.data[0].shape
```

Comment:
Should it be like this?
```suggestion
        if self.s._lazy:
            new_signal.compute()
            zeroDragged.compute()
        assert zeroDragged.data.shape == new_signal.data.shape
        np.testing.assert_allclose(zeroDragged.data[0], new_signal.data[0])
```

Reply:
So I think the problem is actually slicing the lazy dataset which explains the different behavior for lazy/ non-lazy signals. 

Slicing the dataset causes numpy to try to cast to a non object so somewhere in the task graph is a casting step which needs to be suppressed or reversed

Reply:
I am trying to figure out what should be the desired behaviour first: with the suggested changes above, both cases (lazy and non-lazy) fail but with a different failures. Without the changes it only fails for lazy.

I didn't look any further than that! 😉